ui/components/app/menu-bar/account-options-menu.js

```diff
@@ -2,26 +2,46 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import { useHistory } from 'react-router-dom';
 import { useDispatch, useSelector } from 'react-redux';
+import { getAccountLink } from '@metamask/etherscan-link';
 
 import { showModal } from '../../../store/actions';
 import { CONNECTED_ROUTE } from '../../../helpers/constants/routes';
 import { Menu, MenuItem } from '../../ui/menu';
-import getAccountLink from '../../../helpers/utils/account-link';
 import {
   getCurrentChainId,
   getCurrentKeyring,
   getRpcPrefsForCurrentProvider,
   getSelectedIdentity,
 } from '../../../selectors';
 import { useI18nContext } from '../../../hooks/useI18nContext';
-import { useMetricEvent } from '../../../hooks/useMetricEvent';
+import {
+  useMetricEvent,
+  useNewMetricEvent,
+} from '../../../hooks/useMetricEvent';
 import { getEnvironmentType } from '../../../../app/scripts/lib/util';
 import { ENVIRONMENT_TYPE_FULLSCREEN } from '../../../../shared/constants/app';
 
 export default function AccountOptionsMenu({ anchorElement, onClose }) {
   const t = useI18nContext();
   const dispatch = useDispatch();
   const history = useHistory();
+
+  const keyring = useSelector(getCurrentKeyring);
+  const chainId = useSelector(getCurrentChainId);
+  const rpcPrefs = useSelector(getRpcPrefsForCurrentProvider);
+  const selectedIdentity = useSelector(getSelectedIdentity);
+  const { address } = selectedIdentity;
+  const addressLink = getAccountLink(address, chainId, rpcPrefs);
+
+  const getBlockExplorerDomain = () => {
+    if (rpcPrefs?.blockExplorerUrl) {
+      return new URL(addressLink)?.hostname;
+    } else if (['0x1', '0x3', '0x4', '0x5', '0x2a'].includes(chainId)) {
+      return 'etherscan.io';
+    }
+    return '';
+  };
```

Comment:
```suggest
const blockExplorerUrl = new URL(addressLink)?.hostname;
```

This is what I meant, the logic duplicated extends to the '0x1'...etc checks. Sorry if I wasn't clear enough.

Reply:
this is what I kind of meant, you don't need any of this logic just use the hostname. Its even fine if we get rinkeby.etherscan.io that's valuable information to include in the events doesn't have to be normalized so that all testnets say 'etherscan.io'

Reply:
Sorry for not being clear enough in my original feedback @adonesky1

Reply:
Yup 😬   my bad. This makes more sense 👍